nonlinear_solid_mechanics/nonlinear_solid_mechanics.cc

```diff
@@ -0,0 +1,1495 @@
+/* ---------------------------------------------------------------------
+ * Copyright (c) 2020 by the preCICE authors
+ *
+ * This file is part of the dealii-adapter for the coupling library
+ * preCICE. Parts of this program are based on deal.II tutorial programs.
+ *
+ * This adapter is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version. The full text of the license can
+ * be found in the file LICENSE in the precice/dealii-adapter repository.
+ * Parts of this program are based on the previous work of Jean-Paul Pelteret
+ * and Andrew McBride in their deal.II code gallery program 'Quasi-Static
+ * Finite-Strain Compressible Elasticity'
+ * ---------------------------------------------------------------------
+ *
+ * Author: David Schneider 2020
+ */
+#include <deal.II/base/function.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/quadrature_point_data.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/base/work_stream.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping_q_eulerian.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/block_sparse_matrix.h>
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/precondition_selector.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/sparse_direct.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <deal.II/physics/elasticity/kinematics.h>
+#include <deal.II/physics/elasticity/standard_tensors.h>
+#include <deal.II/physics/transformations.h>
+
+#include <fstream>
+#include <iostream>
+
+#include "include/coupling_functions.h"
+#include "include/parameter_handling.h"
+#include "include/time.h"
+#include "precice/SolverInterface.hpp"
+
+namespace adapter
```

Comment:
Why this namespace? Logically this code is now the "solver" and calls the "adapter"? Right?